### PR TITLE
Integration with BE

### PR DESCRIPTION
**Pull request by Vsevolod1020**

### CODE REVIEW

File: src/app/modules/admin/components/profile-content/profile-content.component.ts [hunk 1]
```diff
@@ -46,12 +50,17 @@ export class ProfileContentComponent implements OnInit, OnDestroy {
             ]),
         });
         this.store.dispatch(LoadAdminActions.getAdminInfo());
+        this.selectId$.subscribe(id => {
```
Comment (glebek2h) [inline src/app/modules/admin/components/profile-content/profile-content.component.ts]: don't do this

File: src/app/modules/admin/components/profile-content/profile-content.component.ts [hunk 2]
```diff
@@ -77,7 +86,7 @@ export class ProfileContentComponent implements OnInit, OnDestroy {
     }
 
     onRemoveFile() {
-        this.store.dispatch(LoadAdminActions.removeProfileAvatar());
+        // this.store.dispatch(LoadAdminActions.removeProfileAvatar());
```
Comment (glebek2h) [inline src/app/modules/admin/components/profile-content/profile-content.component.ts]: remove comments

File: src/app/modules/admin/state/effects/admin.effects.ts [hunk 3]
```diff
@@ -11,62 +10,63 @@ export class AdminEffects {
...
-                }),
-            ),
-            map(removeDataResponse =>
-                AdminActions.removeProfileAvatarSuccess({
-                    removeDataResponse,
-                }),
-            ),
-        );
-    });
+    // uploadProfileAvatar$ = createEffect(() => {
```
Comment (glebek2h) [inline src/app/modules/admin/state/effects/admin.effects.ts:41]: add comment with link to trello task

File: src/app/modules/admin/state/effects/orders.effects.ts [hunk 4]
```diff
@@ -20,6 +19,19 @@ export class OrdersEffects {
         );
     });
 
+    deleteOrder$ = createEffect(() => {
+        return this.actions$.pipe(
+            ofType(OrdersActions.deleteOrder),
+            mergeMap(({orderId}) =>
```
Comment (glebek2h) [inline src/app/modules/admin/state/effects/orders.effects.ts]: check if switchMap is ok here

File: src/app/modules/admin/state/reducers/admin.reducer.ts [hunk 5]
```diff
@@ -5,9 +5,12 @@ import { AdminInfoState } from '../admin.state';
 export const adminFeatureKey = 'profile-info';
 
 export const initialState: AdminInfoState = {
-    name: null,
-    email: null,
-    avatar: null,
+    profile: {
```
Comment (glebek2h) [inline src/app/modules/admin/state/reducers/admin.reducer.ts]: replace with null

File: src/app/modules/admin/state/reducers/admin.reducer.ts [hunk 6]
```diff
@@ -31,20 +32,29 @@ export const reducer = createReducer<AdminInfoState>(
...
-            avatar: null,
-        };
-    }),
-    on(AdminActions.uploadProfileAvatarSuccess, (state, action) => {
-        return {
-            ...state,
-            avatar: action.uploadAvatar,
+            profile: {
+                ...state,
+                _id: action.updatedData._id,
```
Comment (glebek2h) [inline src/app/modules/admin/state/reducers/admin.reducer.ts]: replace with ...action.updatedData

File: src/app/modules/admin/state/reducers/admin.reducer.ts [hunk 7]
```diff
@@ -31,20 +32,29 @@ export const reducer = createReducer<AdminInfoState>(
...
+                avatar: action.updatedData.avatar,
+                name: action.updatedData.name,
+                email: action.updatedData.email
+            }
         };
     }),
+    // on(AdminActions.removeProfileAvatarSuccess, state => {
```
Comment (glebek2h) [inline src/app/modules/admin/state/reducers/admin.reducer.ts:34]: add link to trello task

File: src/app/modules/admin/state/reducers/orders.reducer.ts [hunk 8]
```diff
@@ -17,4 +18,17 @@ export const reducer = createReducer(
             orders: action.orders,
         };
     }),
+    on(OrdersActions.deleteOrder, (state, action) => {
```
Comment (glebek2h) [inline src/app/modules/admin/state/reducers/orders.reducer.ts]: remove from reducers

File: src/app/modules/admin/state/reducers/orders.reducer.ts [hunk 9]
```diff
@@ -6,6 +6,7 @@ export const ordersFeatureKey = 'orders-list';
 
 export const initialState: OrdersState = {
     orders: [],
+    orderId: null,
```
Comment (glebek2h) [inline src/app/modules/admin/state/reducers/orders.reducer.ts]: remove this field from state

File: src/app/modules/admin/state/selectors/admin.selectors.ts [hunk 10]
```diff
@@ -7,10 +7,15 @@ export const selectAdminFeature =
 
 export const selectName = createSelector(
     selectAdminFeature,
```
Comment (glebek2h) [inline src/app/modules/admin/state/selectors/admin.selectors.ts]: selectProfile

File: src/app/modules/admin/state/selectors/admin.selectors.ts [hunk 11]
```diff
@@ -7,10 +7,15 @@ export const selectAdminFeature =
...
-    (state: AdminInfoState) => state.name,
+    (state: AdminInfoState) => state.profile.name,
 );
 
 export const selectAvatar = createSelector(
     selectAdminFeature,
-    (state: AdminInfoState) => state.avatar,
+    (state: AdminInfoState) => state.profile.avatar,
```
Comment (glebek2h) [inline src/app/modules/admin/state/selectors/admin.selectors.ts]: export const selectProfile = createSelector(
    selectAdminFeature,
    (state: AdminInfoState) => state.profile,
);

export const selectName = createSelector(
    selectProfile 
    (state) => state.name, // todo add interface
);

File: src/app/services/orders-service/orders-server.service.ts [hunk 12]
```diff
@@ -11,9 +11,15 @@ import * as serviceConstants from '../service.constants';
...
 
-    getOrders(): Observable<APIResponse<Orders>> {
-        return this.http.get<APIResponse<Orders>>(
+    getOrders(): Observable<APIResponseOrders<Orders>> {
+        return this.http.get<APIResponseOrders<Orders>>(
             `${environment.api_url}${serviceConstants.orders}`,
         );
     }
+
+    deleteOrder(id: string) {
```
Comment (glebek2h) [inline src/app/services/orders-service/orders-server.service.ts]: ret type

File: src/app/modules/admin/components/profile-content/profile-content.component.ts (hunk 1 above)
Comment (Vsevolod1020) [reply src/app/modules/admin/components/profile-content/profile-content.component.ts]: How I can grab id and put it in dataObj below. I have to assign it to a variable anyway.

Comment (glebek2h) [reply src/app/modules/admin/components/profile-content/profile-content.component.ts]: no
at least you can do it inside of selected subscription

File: src/app/modules/admin/state/admin.model.ts [hunk 13]
```diff
@@ -17,26 +18,42 @@ export interface Orders {
...
-    orders: Array<Orders>
-  }
+export interface APIResponseOrders<T> {
+    orders: Array<Orders>;
+}
+export interface APIResponseProfile<T> {
+    profile: Admin;
+}
+
+export interface SelectName {
```
Comment (glebek2h) [inline src/app/modules/admin/state/admin.model.ts]: remove all this "Select*" interfaces

File: src/app/modules/admin/state/selectors/admin.selectors.ts [hunk 14]
```diff
@@ -1,16 +1,27 @@
...
+export const selectName = createSelector(
+    selectProfile,
+    (state: SelectName) => state.name,
 );
 
 export const selectAvatar = createSelector(
-    selectAdminFeature,
-    (state: AdminInfoState) => state.avatar,
+    selectProfile,
+    (state: SelectAvatar) => state.avatar,
```
Comment (glebek2h) [inline src/app/modules/admin/state/selectors/admin.selectors.ts]: here should be Profile type

**Pull request merged**